### PR TITLE
Voltron the builder / One Builder to Rule Them All / Make Builder Great Again

### PR DESCRIPTION
**Pull request by 121watts**
- [x] CHANGELOG.md updated
  - [x] Rebased/mergable
  - [x] Tests pass
  - [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

Connect #1169

### CODE REVIEW

**Comment by lukevmorris** [comment]
Holy code

File: ui/src/influxql/ast.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
+import _ from 'lodash'
```
Comment (lukevmorris) [inline ui/src/influxql/ast.js]: Was this all written from scratch, or is it a port of an existing algorithm?

File: ui/src/data_explorer/components/View.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
+import React, {PropTypes} from 'react'
+
+import Table from './Table'
+import AutoRefresh from 'shared/components/AutoRefresh'
+import LineGraph from 'shared/components/LineGraph'
+import SingleStat from 'shared/components/SingleStat'
+const RefreshingLineGraph = AutoRefresh(LineGraph)
+const RefreshingSingleStat = AutoRefresh(SingleStat)
+
+const View = ({
```
Comment (lukevmorris) [inline ui/src/data_explorer/components/View.js]: I know this is scoped within `data_explorer`, but can we rename this to something more specific to its function? `View` is a bit too general and doesn't help to provide context.

File: ui/src/shared/components/AutoRefresh.js [hunk 3]
```diff
@@ -150,3 +148,5 @@ export default function AutoRefresh(ComposedComponent) {
 
   return wrapper
 }
+
+export default AutoRefresh
```
Comment (lukevmorris) [inline ui/src/shared/components/AutoRefresh.js]: What's the argument for this pattern over the following? Should we add this type of thing to our central style guide so we can converge on a style?

```
export const AutoRefresh = (ComposedComponent) => {
```

File: ui/src/shared/actions/app.js [hunk 4]
```diff
@@ -20,3 +20,8 @@ export const setAutoRefresh = (milliseconds) => ({
     milliseconds,
   },
 })
+
+export const noop = () => ({
```
Comment (lukevmorris) [inline ui/src/shared/actions/app.js]: Nice.

File: ui/src/influxql/ast.js (hunk 1 above)
Comment (cryptoquick) [inline ui/src/influxql/ast.js]: haha, nope, I wrote it. pairing with goller and watts, of course. we powered through it, took us like, five hours, lol.

File: ui/src/shared/actions/app.js (hunk 4 above)
Comment (cryptoquick) [inline ui/src/shared/actions/app.js]: lol, noop, we has it!

**Comment by cryptoquick** [comment]
LGTM 👍

**Comment by lukevmorris** [comment]
BOOM

**Pull request merged**